consensus/src/experimental/buffer_manager.rs

```diff
@@ -164,6 +167,60 @@ impl StateManager {
         }
     }
 
+    async fn process_ordered_blocks(
+        &mut self,
+        ordered_blocks: OrderedBlocks,
+    ) -> anyhow::Result<()> {
+        let OrderedBlocks {
+            blocks,
+            finality_proof,
+            callback,
+        } = ordered_blocks;
+
+        // push blocks to buffer
+        for eb in blocks.iter() {
+            self.buffer
+                .push_back(BufferItem::Block(Arc::new(eb.clone())));
+        }
+
+        self.buffer
+            .push_back(BufferItem::FinalityProof(Box::new(FinalityProofItem(
+                finality_proof.clone(),
+                BTreeMap::new(),
+                callback,
+            ))));
+
+        // send blocks to execution phase
+        self.execution_phase_tx
+            .send(ExecutionRequest {
+                blocks: blocks
+                    .iter()
+                    .map(|eb| eb.block().clone())
+                    .collect::<Vec<Block>>(),
+            })
+            .await?;
+        Ok(())
+    }
+
+    async fn process_reset_req(&mut self, reset_event: ResetRequest) -> anyhow::Result<()> {
```

Comment:
as we discussed, reset request (should be renamed to sync request too) should carry a LedgerInfo and we don't necessarily pop everything. Instead we pop until the ledger info block.

Reply:
resolved